mongomock/collection.py

```diff
@@ -1699,6 +1699,11 @@ def batch_size(self, count):
     def close(self):
         pass
 
+    def hint(self, unused_hint):
+        if self._emitted:
+            raise InvalidOperation('cannot set options after executing query')
+        return self
```

Comment:
Please also ensure the following
> You cannot use hint() if the query includes a $text query expression.
https://docs.mongodb.com/manual/reference/method/cursor.hint/#behavior

Reply:
I've added a TODO because that would require to implement text indexes.